### PR TITLE
[DC-1120]Update Snapshot Builder Settings to include Dataset Table Name and Relationships

### PR DESCRIPTION
**Pull request by rjohanek**
## Summary of changes:
Update the structure of the snapshot builder settings to match the changes on the backend
Update the structure of the snapshot access request to match the changes on the backend
Follow the changes through with some renaming and updating usages

### Testing strategy
I ran this branch connected to local TDR on branch DC-1120-update-settings. I followed these steps:
1. Use the set up resources script to get a full view snapshot locally, off which I can make snapshot access requests.
2. Update the settings on that full view snapshot to be the new settings shown in this PR.
3. On local terra-ui, go through the researcher flow to submit a snapshot access request. Below is a video of the flow and the json of the submitted snapshot access request. It includes the Drug and Condition tables.

Video of creating a snapshot access request: 
https://github.com/user-attachments/assets/7e8d7f5f-ecc5-437d-b5e8-4165163dd8d1

Resulting snapshot access request response: 
```
{
    "id": "6f8a1c2f-e1c9-4af2-a7ad-bb7fbc9bb016",
    "datasetId": null,
    "sourceSnapshotId": "f5555e71-f988-4119-ab22-20bbaf37a1cc",
    "snapshotName": "",
    "snapshotResearchPurpose": "",
    "snapshotSpecification": {
        "cohorts": [
            {
                "name": "cohort",
                "criteriaGroups": [
                    {
                        "name": "Group 4",
                        "criteria": [
                            {
                                "kind": "domain",
                                "id": 10,
                                "conceptId": 4187458
                            },
                            {
                                "kind": "domain",
                                "id": 10,
                                "conceptId": 4163872
                            },
                            {
                                "kind": "range",
                                "id": 0,
                                "low": 1911,
                                "high": 2005
                            },
                            {
                                "kind": "list",
                                "id": 1,
                                "values": [
                                    8532
                                ]
                            }
                        ],
                        "mustMeet": true,
                        "meetAll": false
                    }
                ]
            }
        ],
        "outputTables": [
            {
                "name": "Drug",
                "columns": []
            },
            {
                "name": "Condition",
                "columns": []
            }
        ]
    },
    "createdBy": "rjohanekdev@gmail.com",
    "status": "SUBMITTED",
    "createdDate": "2024-07-16T20:53:21.027691Z",
    "statusUpdatedDate": null,
    "flightid": null,
    "createdSnapshotId": null
}
```
4. On local tdr swagger, approve the request.
5. On local tdr swagger, create a snapshot using the byRequestId mode and the id of the snapshot access request id. Using this payload:
```
{
  "name": "unused_overwritten",
  "description": "testing settings changes with new backend changes",
  "contents": [
    {
      "datasetName": "unused_overwritten",
      "mode": "byRequestId",
      "requestIdSpec": {
        "snapshotRequestId": "6f8a1c2f-e1c9-4af2-a7ad-bb7fbc9bb016"
      }
    }
  ]
}
```
6. On local tdr swagger, use the querySnapshotDataById endpoint to check that the newly created snapshot has the selected tables (Drug and Condition) populated as well as the person and concept tables. Below is a video that shows what data was included in this snapshot and that other data, such as Procedure was not included.
Video showing snapshot data: 

https://github.com/user-attachments/assets/f4c2e595-d024-4f4c-97d1-16f315e88fcf

### Notes
Once this is merged, the settings on dev and prod will need to be updated for the product to work.

### CODE REVIEW

**Review by snf2ye — APPROVED** [review]

File: src/dataset-builder/DatasetBuilder.test.ts [hunk 1]
```diff
@@ -133,10 +132,7 @@ describe('DatasetBuilder', () => {
   const renderConceptSetSelector = () =>
     render(
       h(ConceptSetSelector, {
-        conceptSets: [
-          ..._.map(convertDomainOptionToConceptSet, testSettings.domainOptions),
-          ...testSettings.datasetConceptSets,
-        ],
+        conceptSets: [...(testSettings.datasetConceptSets ?? [])],
```
Comment (s-rubenstein) [inline src/dataset-builder/DatasetBuilder.test.ts]: Do we still need the `??` statement? Or can this just be: `conceptSets: testSettings.datasetConceptSets`?

File: src/dataset-builder/DatasetBuilder.ts [hunk 2]
```diff
@@ -641,14 +635,9 @@ export const DatasetBuilderView: React.FC<DatasetBuilderProps> = (props) => {
...
-  const [selectedValues, setSelectedValues] = useState([] as RequiredHeaderAndValues<DatasetBuilderValue>[]);
+  const [selectedColumns, setSelectedColumns] = useState([] as RequiredHeaderAndValues<DatasetBuilderValue>[]);
   const conceptSets =
-    snapshotBuilderSettings.status === 'Ready'
-      ? [
-          ..._.map(convertDomainOptionToConceptSet, snapshotBuilderSettings.state.domainOptions),
-          ...(snapshotBuilderSettings.state.datasetConceptSets ?? []),
-        ]
-      : [];
+    snapshotBuilderSettings.status === 'Ready' ? [...(snapshotBuilderSettings.state.datasetConceptSets ?? [])] : [];
```
Comment (s-rubenstein) [inline src/dataset-builder/DatasetBuilder.ts]: Same question, do we need the wrapping array now?

File: src/dataset-builder/TestConstants.ts [hunk 3]
```diff
@@ -80,38 +80,69 @@ export const testSnapshotBuilderSettings = (): SnapshotBuilderSettings => ({
...
+      },
     },
-    {
-      values: ['demographics column 1', 'demographics column 2'],
-      name: 'Person',
-    },
-  ],
-  datasetConceptSets: [
     {
       name: 'Demographics',
```
Comment (s-rubenstein) [inline src/dataset-builder/TestConstants.ts]: Optional: My understanding is that we were pulling `Demographics` - should we exclude that here? (This is test data so it doesn't _really_ matter, so do if you want)

**Review by s-rubenstein — COMMENTED** [review]
This looks reasonable. One suggested change.

Also, its interesting that we need to define the relationships here. I would think we could mostly not include them and look them up in the backend when we need them? But it seems fine either way.

File: src/dataset-builder/TestConstants.ts (hunk 3 above)
Comment (rjohanek) [reply src/dataset-builder/TestConstants.ts]: yes good idea

File: src/dataset-builder/DatasetBuilder.test.ts (hunk 1 above)
Comment (rjohanek) [reply src/dataset-builder/DatasetBuilder.test.ts]: no good catch

**Comment by rjohanek** [comment]
> Also, its interesting that we need to define the relationships here. I would think we could mostly not include them and look them up in the backend when we need them? But it seems fine either way.

What do you mean? I included the relationships here just so the type would match when the settings are retrieved from the backend so it doesn't error. Even though you're right they aren't used on the UI currently. I could make a separate UI type that excludes the relationships and convert it before it gets passed around.

**Review by s-rubenstein — APPROVED** [review]
@raejohanek, mostly the fact that we are ending up defining the relationships twice, right? Once in the dataset schema, and then once in the settings. Technically we could probably look up the relationships in the schema based on the table name and then use those. That said, I think it is okay to have some duplicate information here.

**Pull request merged**